File: code_samples/elixir/functional_currying.ex
# Functional: Currying
# AI/ML Training Sample

defmodule Currying do
  defstruct data: ""
  
  def new(), do: %Currying{}
  
  def process(%Currying{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Currying{data: data}), do: data
  
  def validate(%Currying{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Currying.new()
updated = Currying.process(instance, "example")
IO.puts("Data: " <> Currying.get_data(updated))
IO.puts("Valid: " <> to_string(Currying.validate(updated)))

File: code_samples/haskell/functional_currying.hs
-- Functional: Currying
-- AI/ML Training Sample

module Currying where

data Currying = Currying {
    getData :: String
} deriving (Show, Eq)

process :: Currying -> String -> Currying
process obj input = obj { getData = input }

validate :: Currying -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Currying { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
